components/VM_Arm/src/main.c

```diff
@@ -865,7 +865,12 @@ static int load_linux(vm_t *vm, const char *kernel_name, const char *dtb_name, c
             paths = camkes_dtb_get_node_paths(&num_paths);
         }
 
-        int dtb_fd = open(linux_image_config.dtb_base_name, 0);
+        int dtb_fd = -1;
+
+        /* No point checking the file server if the string is empty! */
+        if (linux_image_config.dtb_base_name[0] != '\0') {
```

Comment:
Yes, please add this check. Otherwise one could also argue that `open()` is supposed to handle these cases well and it's a user config error if `CONFIG_VM_DTB_FILE` is enabled and then not used. With the change we might eventually argue that `CONFIG_VM_DTB_FILE` could be removed, so setting this attribute or leaving it empty is the way how the behavior is defined.

Reply:
Added a null check. Always good to check a pointer before referencing it :)